### DISCUSSION TITLE
Is there a Setting to auto-capture text when highlighting and populate the annotation's contents with the highlighted text?

### DISCUSSION THREAD

**Discussion by shern15** [sumatrapdfreader/sumatrapdf#4359] [Q&A] [answered]
Hi! First off, thank you so much for all the hard work on this application! I just started using it and it is great!

I highlight a lot of text in my PDF textbooks and right now the only way I can find to populate a Highlight annotation's contents with the text I am highlighting is to CTRL + C the text before highlighting, then A, then CTRL + V into the Contents box when the Edit Annotation window appears.

I was wondering if there was some kind of Setting somewhere that I can enable so that I can just press 'a' (or whatever shortcut for the desired highlight color) and have the Contents box of the annotation automatically populated with the highlighted text?

I  am using Prerelease 16388.

Thank you for your time!

**Comment by kjk** [comment, marked as answer]
Not currently. We copy the content to clipboard so you can `Ctrl + V`.

I just added ability to specify arguments to commands (see https://github.com/sumatrapdfreader/sumatrapdf/issues/4346).

I can add a flag `autopopulatecontent` to `CmdCreateAnnotHighlight` command and then you could re-bind `a` key to:
```
Shortcuts [
	[
		Cmd = CmdCreateAnnotHighlight #00ff00 autopopulatecontent
		Key = a
	]
]
```

Open an issue if you want that.

**Reply by shern15** [reply]
I have opened issue https://github.com/sumatrapdfreader/sumatrapdf/issues/4360 from this discussion.

Thank you!

**Comment by H5820121** [comment]
@Krzysztof Kowalczyk
"I just added ability to specify arguments to commands"
Has the option to set a keyboard shortcut for ExternalViewers already been
introduced to the pre-release version, what is the command?

‫בתאריך יום ה׳, 4 ביולי 2024 ב-20:19 מאת ‪shern15‬‏ <‪
***@***.***‬‏>:‬
